carto/auth.py

```diff
@@ -169,4 +175,51 @@ def send(self, relative_path, http_method, **requests_args):
                                                       http_method,
                                                       **requests_args)
         except Exception as e:
-            raise CartoException(e)
\ No newline at end of file
+            raise CartoException(e)
+
+
+class AuthAPIClient(_UsernameGetter, _BaseUrlChecker, BasicAuthClient):
+    """
+    This class provides you with authenticated access to CARTO's APIs using
+    your API key at Basic authentication header, as provided by Auth API.
+
+    Auth API is still under development. You might want to take a look at
+    APIKeyAuthClient for missing features or an stable API.
+
+    You can find your API key by clicking on the API key section of the user
+    dropdown menu
+    """
+
+    def __init__(self, base_url, api_key, organization=None, session=None):
+        """
+        Init method
+
+        :param base_url: Base URL. API endpoint paths will always be relative
+        to this URL
+        :param api_key: API key
+        :param organization: For enterprise users, organization user belongs to
+        :param session: requests' session object
+        :type api_key: str
+        :type organization: str
+
+        :return:
+        """
+        self.organization = organization
+        self.api_key = api_key
+        base_url = self.check_base_url(base_url)
+        self.username = self.get_user_name(base_url)
+
+        super(AuthAPIClient, self).__init__(self.username, api_key, base_url, session=session)
+
+    def is_valid_api_key(self):
+        """
+        Checks validity. Right now, an API key is considered valid if it
+        can list user API keys and the result contains that API key.
+        This might change in the future.
+
+        :return: True if the API key is considered valid for current user.
+        """
+        res = self.send('/api/v3/api_keys', 'get')
+        return \
+            res.status_code == 200 and \
+            self.api_key in map(lambda ak: ak['token'], res.json()['result'])
```

Comment:
It's generally considered pythonic to favour list comprehensions over `map`/`filter`. In this case:

`self.api_key in (ak['token'] for ak in res.json()['result'])`

This is also using a generator (the comprehension is surrounded by `()` instead oof `[]`), so it will stop processing as soon as it find a match.

https://docs.python.org/2/howto/functional.html#generator-expressions-and-list-comprehensions